regression/benchmark/benchmarks.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MPL-2.0
+set -e
+
+# Ensure all dependencies are installed
+command -v jq >/dev/null 2>&1 || { echo >&2 "jq is not installed. Aborting."; exit 1; }
+command -v gcc >/dev/null 2>&1 || { echo >&2 "gcc is not installed. Aborting."; exit 1; }
+
+# Script directory
+BENCHMARK_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+# Kernel image 
+KERNEL_DIR="/root/dependency/linux_kernel"
+LINUX_KERNEL="${KERNEL_DIR}/vmlinuz-5.15.0-105-generic"
+
+# Generate init script for Linux cases
+generate_init_script() {
+    local benchmark="$1"
+    local init_script=$(cat <<EOF
+#!/bin/sh
+
+echo "Running ${benchmark}"
+chmod +x /benchmark/${benchmark}/run.sh
+/benchmark/${benchmark}/run.sh
+
+poweroff -f
+EOF
+)
+    echo "$init_script"
+}
+
+prepare_benchmark() {
+    echo "Preparing benchmark environment..."
+    mkdir -p "/benchmark"
+
+    # Copy sysbench
+    echo "Copying sysbench to /benchmark..."
+    cp /usr/local/benchmark/sysbench/bin/sysbench /benchmark
```

Comment:
The current code organization still fails to keep benchmark-specific code such as the one specific to `sysbench` under its own directory. This script still hardcodes information about `sysbench` and `getpid`.

Similarly, `benchmark/config.json` is also a centralized place that keep information about every benchmark. Ideally, for every new benchmark, we want to add only a new directory. No more, no less.

Reply:
For benchmark preparation, we can remove it because now the `Makefile` is responsible for moving and compiling benchmarks.

I will also replace `benchmark/config.json` with `<benchmark>/threshold.json` for every benchmark.